packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -413,11 +413,57 @@ class IOSDevice extends Device {
       }
 
       _logger.printTrace('Application launched on the device. Waiting for observatory url.');
-      final Timer timer = Timer(discoveryTimeout ?? const Duration(seconds: 30), () {
-        _logger.printError('iOS Observatory not discovered after 30 seconds. This is taking much longer than expected...');
+
+      final int defaultTimeout = interfaceType == IOSDeviceConnectionInterface.network ? 45 : 30;
+      final Timer timer = Timer(discoveryTimeout ?? Duration(seconds: defaultTimeout), () {
+        _logger.printError('iOS Observatory not discovered after $defaultTimeout seconds. This is taking much longer than expected...');
+
+        // If debugging with a wireless device and the timeout is reached, remind the
+        // user to allow local network permissions.
+        if (interfaceType == IOSDeviceConnectionInterface.network) {
+          _logger.printError(
+            'Click "Allow" to the prompt asking if you would like to find and connect devices on your local network. '
+            'This is required for wireless debugging. If you selected "Don\'t Allow", '
+            'you can turn it on in Settings > Your App Name > Local Network. '
+            "If you don't see your app in the Settings, uninstall the app and rerun to see the prompt again."
+          );
+        }
+
         iosDeployDebugger?.pauseDumpBacktraceResume();
       });
-      final Uri? localUri = await observatoryDiscovery?.uri;
+
+      Uri? localUri;
+      if (interfaceType == IOSDeviceConnectionInterface.network) {
+        // Wait for iOS Observatory to start up.
+        await observatoryDiscovery?.uri;
+
+        // If observatory url is not found within 2.2 seconds, change the status
```

Comment:
This caller shouldn't have to know the guts of `MDnsObservatoryDiscovery` to know it's a 2 second timeout.  Instead, try adding the lookup duration parameter into `MDnsObservatoryDiscovery`, and then add 200 ms to it out here.